Checksum `from` address to avoid addresses comparison fail

jcarpanelli opened the pull request:
Fixes https://github.com/zeppelinos/zos/issues/648.

spalladino commented on packages/lib/src/artifacts/ZWeb3.ts:
```diff
@@ -61,6 +65,14 @@ export default class ZWeb3 {
     return (await ZWeb3.accounts())[0];
   }
 
+  public static toChecksumAddress(address: string): string {
```
Add a throw if checksum is incorrect. Logic should then be:

- If address seems checksummed (ie has at least an uppercase letter), validate it
  - If it is incorrect, throw
  - If it is correct, go
- If address has no uppercase letters, show a warning, and checksum it ourselves

jcarpanelli commented:
@spalladino, already implemented your suggestions. I also removed all `.toLowerCase()` I found, as `truffle-hdwallet-provider` does not complain if you provide a non-lowercased address while using private keys instead of a mnemonic.
Also, I added some tests :slightly_smiling_face:.

spalladino commented on packages/lib/test/src/artifacts/ZWeb3.test.js:
```diff
@@ -189,6 +189,34 @@ contract('ZWeb3', accounts => {
+                const checksummed = ZWeb3.toChecksumAddress(address);
+              } catch(error) {
+                error.message.should.match(/is not a valid Ethereum address or it has not been checksummed correctly/);
+              }
```
Note that this test does not actually test that it fails! If the call does not throw, the catch is never executed, and the test succeeds. Look for the `expects(fn).to.throw()` method [in chai](https://www.chaijs.com/api/bdd/#method_throw) to implement this.

jcarpanelli replied:
Fixed!
